lib/libp2p_port.ex

```diff
@@ -370,11 +421,55 @@ defmodule LambdaEthereumConsensus.Libp2pPort do
     {:noreply, state |> Map.put(:requests, new_requests)}
   end
 
-  def handle_cast({:add_block, block}, state) do
-    PendingBlocks.add_block(block)
+  @impl GenServer
+  def handle_cast({:add_blocks, {first_slot, last_slot}, blocks}, state) do
+    n_blocks = length(blocks)
+    missing = last_slot - first_slot + 1 - n_blocks
+
+    Logger.info(
+      "[Optimistic Sync] Range #{first_slot} - #{last_slot} downloaded successfully, with #{n_blocks} blocks and #{missing} missing."
+    )
+
+    Enum.each(blocks, &PendingBlocks.add_block/1)
+    new_state = Map.update!(state, :blocks_remaining, fn n -> n - n_blocks - missing end)
+
+    if new_state.blocks_remaining > 0 do
+      Logger.info("[Optimistic Sync] Blocks remaining: #{new_state.blocks_remaining}")
+      {:noreply, new_state}
+    else
+      Logger.info("[Optimistic Sync] Sync completed. Subscribing to gossip topics.")
+
+      [
+        LambdaEthereumConsensus.P2P.Gossip.BeaconBlock,
+        LambdaEthereumConsensus.P2P.Gossip.BlobSideCar,
+        LambdaEthereumConsensus.P2P.Gossip.OperationsCollector
+      ]
+      |> Enum.flat_map(&topics_for_module/1)
+      |> Enum.reduce(new_state, fn {module, topic}, state ->
+        command = %Command{c: {:subscribe, %SubscribeToTopic{name: topic}}}
+        send_data(state.port, Command.encode(command))
+        add_subscriber(state, topic, module)
+      end)
+      |> then(fn state -> {:noreply, state} end)
+    end
```

Comment:
I think this function is doing too many things. Maybe we can move this to a new `subscribe_to_gossip_topics` function?